contracts/zxp/game/seasons/mechanics/SecretRewards.sol

```diff
@@ -88,6 +93,12 @@ contract SecretRewards is ObjectRegistryClient, ISecretRewards {
         emit SecretCommitted(nonce, secretHash);
     }
 
+    // this whole flow seems redundant. an user that has the hash
+    // (that is available on the contract by the time a user commits a guess,
+    // he can locally keep hashing his guesses to get the same hash,
+    // once he gets the same hash he (and we) already know that his guess is correct,
+    // so we should be able to reward him at guess commit time and this whole reveal secret flow
+    // seems redundant

```

Comment:
Aw but then, only one person would be able to guess and then everybody else would know that it's the right answer 
without having to do the quest.

Actually there is an issue with the contract related to this though. that is WIP. either salt is needed here, or the nonce needs to be abstracted in the guess. in order to prevent copying.